### DISCUSSION TITLE
Is it inefficient to have multiple processes mount the same volume to different mountpoints under same machine's /tmp/

### DISCUSSION THREAD

**Discussion by myc724** [juicedata/juicefs#270] [Q&A] [answered]
Some part of the flow that I’m building might need to have multiple processes mount the same volume to different mountpoints on the same machine as /tmp/run_dir.pid. I’m wondering how much underneath (JuiceFS+FUSE+Redis:metadata) flow difference (and performance degradation) between the following two scenarios. Thanks!

    On the same machine,
        Scenario #1: multiple processes share the same mountpoint under /tmp/.
        Scenario #2: Each of these processes mount the same volume to their own mountpoints under /tmp/.

I guess, a somewhat related question is, does JuiceFS deploy high or low level FUSE APIs? I’m wondering if JuiceFS deploys high level FUSE APIs, “Scenario #2” could use caching outside of JuiceFS flow (e.g. hardware caching). If any of my elaboration is incorrect, please advise. Thanks!

**Comment by davies** [comment, marked as answer]
JuiceFS is implemented using low level FUSE API.

Each JuiceFS process introduce some overhead, so we'd suggest you to go with #1 to share the same mount point to different applications (processes). If the application process need some isolation, then can still use different directories in JuiceFS.

**Reply by myc724** [reply]
Thanks!
